### PR TITLE
Expose picture.debugDisposed, make sure we do not crash when working with disposed pictures

### PR DESCRIPTION
**Pull request by dnfield**
Fixes https://github.com/flutter/flutter/issues/85471
Fixes https://github.com/flutter/flutter/issues/98434

xref b/217635834

### CODE REVIEW

File: testing/dart/compositing_test.dart [hunk 1]
```diff
@@ -8,6 +8,35 @@ import 'dart:ui';
...
+
+    final SceneBuilder builder = SceneBuilder();
+    if (assertsEnabled) {
+      expect(
+        () => builder.addPicture(Offset.zero, picture),
+        throwsA(isInstanceOf<AssertionError>()),
+      );
+    } else {
+      builder.addPicture(Offset.zero, picture);
+    }
```
Comment (dnfield) [inline testing/dart/compositing_test.dart:33]: Without the changes in scene_builder.cc, this code will cause a segfault.

**Comment by dnfield** [comment]
@godofredoc looks like the recipe or ciyaml is out of whack here?

**Comment by dnfield** [comment]
I thought I had merged to head and hadn't I guess. That's likely the problem.

**Review by yjbanov — APPROVED** [review]
lgtm

**Pull request merged**